pkg/controllers/report/resource/controller.go

```diff
@@ -154,7 +156,10 @@ func (c *controller) updateDynamicWatchers(ctx context.Context) error {
 			delete(c.dynamicWatchers, gvr)
 		} else {
 			logger.Info("start watcher ...", "gvr", gvr)
-			watchInterface, err := c.client.GetDynamicInterface().Resource(gvr).Watch(ctx, metav1.ListOptions{})
+			watchFunc := func(options metav1.ListOptions) (watch.Interface, error) {
+				return c.client.GetDynamicInterface().Resource(gvr).Watch(ctx, metav1.ListOptions{})
+			}
+			watchInterface, err := watchTools.NewRetryWatcher("1", &cache.ListWatch{WatchFunc: watchFunc})
```

Comment:
What is `"1"` here ?

Reply:
Hi @eddycharly , it is supposed to instruct the initial resource version from where the watcher will start watching, however in the implementation it's not really used, as you can see at line 160, the `options` argument of the `func`, which will contains that value, is not used, so the underline `GetDynamicInterface().Resource(gvr).Watch` is still being called as in the previous implementation, which means the new watcher will start following the changes from the time being it is created, pretty much saying the `1` here is just a fake starting version to be compliant with the definition of the `NewRetryWatcher` function.

Reply:
actually after reading through [Efficient detection of changes](https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes), I believe it's better to move the `watch` section to after the `list` section and use as the initial version the version which is returned from the most recent object, I will update the PR shortly.

Reply:
done @eddycharly, I'm now following what is described at [Efficient detection of changes](https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes), by first performing the `list`, getting the latest resource version from the items returned and them using it as the starting point for the `watch`

Reply:
@rodrigorfk shouldn't we always start at "1" and learn as events come in ?

Reply:
I need a bit of time to think more about it but it looks great, maybe for 1.8.3.

Reply:
Hi @eddycharly, I added another commit fixing the logic of getting the `resourceVersion`, there is, indeed, no reason for starting at "1" or to get the value from the individual resources returned in the list operation as I was doing before, the list operation itself provides the latest `resourceVersion` for that operation as part of the `metadata` of the operation itself, as you can see in the example above:
```
GET /api/v1/namespaces/test/pods
---
200 OK
Content-Type: application/json

{
  "kind": "PodList",
  "apiVersion": "v1",
  "metadata": {"resourceVersion":"10245"},
  "items": [...]
}
```
by using this value, the watch operation will always start from the tail position where the list operation ended.

Please have a look again, and sure, please take all time you need to think about it, from my side there is no rush, the issue itself is preventing us from using backgroundScan, however we have the option of disable it via the `--backgroundScan=false` flag.

Reply:
Changes look good, let's get this in !
Thanks @rodrigorfk